internal/tofu/context_plan2_test.go

```diff
@@ -4946,6 +4903,157 @@ resource "test_object" "a" {
 	}
 }
 
+func TestContext2Plan_importIntoNonExistentModule(t *testing.T) {
+	m := testModuleInline(t, map[string]string{
+		"main.tf": `
+import {
+  to = module.mod.test_object.a
+  id = "456"
+}
+
+`,
+	})
+	p := simpleMockProvider()
+	ctx := testContext2(t, &ContextOpts{
+		Providers: map[addrs.Provider]providers.Factory{
+			addrs.NewDefaultProvider("test"): testProviderFuncFixed(p),
+		},
+	})
+	_, diags := ctx.Plan(m, states.NewState(), &PlanOpts{
+		Mode: plans.NormalMode,
+	})
+	if !diags.HasErrors() {
+		t.Fatalf("expected error")
+	}
+	if !strings.Contains(diags.Err().Error(), "Cannot import to non-existent resource address") {
+		t.Fatalf("expected error to be \"Cannot import to non-existent resource address\", but it was %s", diags.Err().Error())
+	}
+}
+
+func TestContext2Plan_importIntoNonExistentConfiguration(t *testing.T) {
```

Comment:
For all these test cases - the behaviour should be the same if config generation was defined. I was wondering if I should run these tests both with and without config generation, but have decided to only run them without config generation for now

Would be interested to hear more opinions on that